scripts/generate_hamilton_input_quant_studio.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+
+import csv
+
+from EPPs.common import InvalidStepError, StepEPP
+
+
+class GenerateHamiltonInputQuantStudio(StepEPP):
+    # define the rows and columns in the input plate (standard 96 well plate pattern)
+    plate_rows = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H']
+    plate_columns = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12']
+    csv_column_headers = ['Input Plate', 'Input Well', 'Output Plate', 'Output Well', 'DNA Volume', 'TE Volume']
+    _max_nb_input_containers = 27
+    _max_nb_output_containers = 1
+
+    def __init__(self, argv=None):
+        """ additional argument required for the location of the Hamilton input file so def __init__ customised."""
+        super().__init__(argv)
+        self.hamilton_input = self.cmd_args.hamilton_input
+
+        assert self.csv_column_headers is not None, 'csv_column_headers needs to be set by the child class'
+        assert self._max_nb_input_containers is not None, 'number of permitted input containers needs to be set ' \
+                                                          'by the child class'
+        assert self._max_nb_output_containers is not None, 'number of permitted output containers needs to be set ' \
+                                                           'by the child class'
+
+    @staticmethod
+    def add_args(argparser):
+
+        argparser.add_argument(
+            '-i', '--hamilton_input', nargs='*', help='Add up to 3 file spaces in LIMS step for Hamilton inputs')
+
+    @staticmethod
+    def write_csv(filename, csv_array):
+        """Write the list of list to the file provided as a csv file"""
+        with open(filename, 'w', newline='') as csvFile:
+            writer = csv.writer(csvFile)
+            writer.writerows(csv_array)
+
+    def generate_csv_dict(self):
+        csv_dict = {}
+
+        for art in self.artifacts:
+            if art.type == 'Analyte':
+                output = self.process.outputs_per_input(art.id, Analyte=True)
+                # the script is only compatible with 1 output for each input i.e. replicates are not allowed
+                if len(output) > 1:
+                    raise InvalidStepError('Multiple outputs found for an input %s. '
```

Comment:
This check can be pushed to `_nb_analytes_per_input=1`

Reply:
updated